utilities/os_utils.py

```diff
@@ -218,30 +223,43 @@ def generate_os_matrix_dict(os_name: str, supported_operating_systems: list[str]
     return os_formatted_list
 
 
-def generate_instance_type_rhel_os_matrix(preferences: list[str]) -> list[dict[str, dict[str, Any]]]:
+def generate_linux_instance_type_os_matrix(os_name: str, preferences: list[str]) -> list[dict[str, dict[str, Any]]]:
     """
-    Generate a list of dictionaries representing the instance type matrix for RHEL OS.
-
+    Generate a list of dictionaries representing the instance type matrix for a Linux OS type.
     Each dictionary represents a specific instance type and its configuration.
 
     Args:
-        preferences (list[str]): A list of preferences for the instance types. Preference format is "rhel-<version>".
+        os_name (str): The name of the OS.
+        preferences (list[str]): A list of preferences for the instance types. Preference format is "<os>.<version>".
 
     Returns:
         list[dict[str, dict[str, Any]]]: A list of dictionaries representing the instance type matrix.
+
     """
-    latest_rhel = f"rhel-{max([preference.split('-')[1] for preference in preferences], key=int)}"
 
+    def _extract_version(pref: str) -> int:
+        if match := re.search(r"\d+", pref):
+            return int(match.group())
+        return None
+
+    def _format_data_source_name(preference_name: str) -> str:
+        match = re.search(rf"{os_name}.*?(\d+)", preference_name)
+        if not match:
+            return preference_name
+        data_source_name = f"{os_name.replace('.', '-')}{match.group(1)}"
+        return data_source_name
+
```

Comment:
Maybe :

def _extract_version(pref: str) -> int | None:
    match = re.search(r"\d+", pref)
    return int(match.group()) if match else None

def _format_data_source_name(preference_name: str) -> str:
    version = _extract_version(preference_name)
    if version is None:
        return preference_name
    return f"{os_name.replace('.', '-')}{version}"   -> that way we save regex duplication?